### PR TITLE
When deleting search results use setLexer only when lexer is SCLEX_NULL

### PR DESCRIPTION
**Pull request by ozone10**
improve/partially fix https://github.com/notepad-plus-plus/notepad-plus-plus/issues/12555

### CODE REVIEW

File: PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp [hunk 1]
```diff
@@ -610,7 +610,10 @@ void Finder::deleteResult()
 	auto end = _scintView.execute(SCI_GETLINEENDPOSITION, lno);
 	if (start + 2 >= end) return; // avoid empty lines
 
-	_scintView.setLexer(L_SEARCHRESULT, LIST_NONE); // Restore searchResult lexer in case the lexer was changed to SCLEX_NULL in GotoFoundLine()
+	if (_scintView.execute(SCI_GETLEXER) == SCLEX_NULL)
+	{
+		_scintView.setLexer(L_SEARCHRESULT, LIST_NONE); // Restore searchResult lexer in case the lexer was changed to SCLEX_NULL in GotoFoundLine()
```
Comment (ozone10) [inline PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:615]: maybe `_scintView.makeStyle(L_SEARCHRESULT, nullptr);` is enough

**Comment by ozone10** [comment]
still the main issue is with folding performance

**Comment by donho** [comment]
@ozone10 
Good job!
But how come it fixes "partially" #12555 ?

**Comment by ozone10** [comment]
@donho 
you can close it as it deal with the main issue.

But since upgrading to scintilla 5, usually "first" folding is really slow, and it can have impact on deleting lines.
I've also tried to commenting out some code related to it in `ScintillaEditView.cpp` (IIRC `runMarkers`) and it helps.
But I am not sure how to handle it, so I only fixed issue with `setLexer`.

**Comment by donho** [comment]
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/b3934afd9210f7f580355c7eb54f263d425ebd43

**Pull request closed without merging**